addons/website_slides/models/slide_channel.py

```diff
@@ -801,15 +801,36 @@ def toggle_active(self):
     def message_post(self, *, parent_id=False, subtype_id=False, **kwargs):
         """ Temporary workaround to avoid spam. If someone replies on a channel
         through the 'Presentation Published' email, it should be considered as a
-        note as we don't want all channel followers to be notified of this answer. """
+        note as we don't want all channel followers to be notified of this answer.
+        Also Make sure that only one review can be posted per course."""
         self.ensure_one()
         if kwargs.get('message_type') == 'comment' and not self.can_review:
             raise AccessError(_('Not enough karma to review'))
         if parent_id:
             parent_message = self.env['mail.message'].sudo().browse(parent_id)
-            if parent_message.subtype_id and parent_message.subtype_id == self.env.ref('website_slides.mt_channel_slide_published'):
+            if parent_message.subtype_id and parent_message.subtype_id == self.env.ref(
+                    'website_slides.mt_channel_slide_published'):
                 subtype_id = self.env.ref('mail.mt_note').id
-        return super(Channel, self).message_post(parent_id=parent_id, subtype_id=subtype_id, **kwargs)
+        message = super().message_post(parent_id=parent_id, subtype_id=subtype_id, **kwargs)
+        if kwargs.get("subtype_xmlid") == "mail.mt_comment":
+            if (
+                self.env["mail.message"].search_count(
+                    [
+                        ("res_id", "=", self.id),
+                        ("author_id", "=", message.author_id.id),
+                        ("model", "=", "slide.channel"),
+                        ("subtype_id", "=", self.env.ref("mail.mt_comment").id),
+                    ],
+                    limit=2,
+                )
+                > 1
+            ):
+                raise ValidationError(_("Only a single review can be posted per course."))
+        if message:
+            rating_value = kwargs.get("rating_value", False)
+            if rating_value and message.author_id == self.env.user.partner_id:
```

Comment:
Is it possible to have a 0 rating? Shoudn't it check for False instead?

Reply:
It's not possible. The value is between 1 and 5.